### PR TITLE
Cleanup Protobuf terminology 

### PR DESCRIPTION
**Pull request by eddyxu**
Discussed in #9

### CODE REVIEW

File: cpp/src/lance/format/metadata.cc [hunk 1]
```diff
@@ -28,37 +41,45 @@ Result<shared_ptr<Metadata>> Metadata::Parse(shared_ptr<::arrow::Buffer> buffer)
...
+  if (it == pb_.batch_offsets().end()) {
+    return ::arrow::Status::IndexError("Row index out of range {} of {}", row_index, len);
   }
-  int32_t bound_idx = std::distance(pb_.chunk_offsets().begin(), it);
+  int32_t bound_idx = std::distance(pb_.batch_offsets().begin(), it);
   assert(bound_idx >= 0);
   bound_idx = std::max(0, bound_idx - 1);
-  int32_t idx_in_chunk = idx - pb_.chunk_offsets(bound_idx);
-  return std::tuple(bound_idx, idx_in_chunk);
+  int32_t idx_in_batch = row_index - pb_.batch_offsets(bound_idx);
```
Comment (changhiskhan) [inline cpp/src/lance/format/metadata.cc:81]: So here we're using `index` and `offset` interchangeably? Or I guess `index` is the "absolute" position and "offset" is relative to the index?

File: cpp/src/lance/format/schema.cc [hunk 2]
```diff
@@ -172,6 +173,13 @@ ::arrow::Status Field::LoadDictionary(std::shared_ptr<::arrow::io::RandomAccessF
   return set_dictionary(dict_arr);
 }
 
+int32_t Field::GetFieldsCount() const {
```
Comment (changhiskhan) [inline cpp/src/lance/format/schema.cc:176]: So this is a flattened count right?

File: cpp/src/lance/format/schema.h [hunk 3]
```diff
@@ -190,6 +193,8 @@ class Field final {
 
   bool RemoveChild(int32_t id);
 
+  int32_t GetFieldsCount() const;
```
Comment (changhiskhan) [inline cpp/src/lance/format/schema.h:196]: can you add a quick code comment to clarify whether this include the field itself

File: cpp/src/lance/format/schema.h [hunk 4]
```diff
@@ -75,6 +75,9 @@ class Schema final {
   /// Top level fields;
   const std::vector<std::shared_ptr<Field>> fields() const { return fields_; }
 
+  /// Count the number of all fields, including nested fields.
```
Comment (changhiskhan) [inline cpp/src/lance/format/schema.h:78]: so this is the number of leaves in the schema tree right?

**Review by changhiskhan — APPROVED** [review]
some nits/questions

File: cpp/src/lance/format/schema.h (hunk 4 above)
Comment (eddyxu) [reply cpp/src/lance/format/schema.h:78]: This is for the size of the entire tree, including parent nodes.

File: cpp/src/lance/format/schema.h (hunk 3 above)
Comment (eddyxu) [reply cpp/src/lance/format/schema.h:196]: Sure will do

File: cpp/src/lance/format/schema.cc (hunk 2 above)
Comment (eddyxu) [reply cpp/src/lance/format/schema.cc:176]: This is getting the node counts of the entire schema tree.

**Pull request merged**